### PR TITLE
Feature/change svg colour

### PR DESCRIPTION
**Pull request by SniraJavas**
clean and finalised code of adding a potion of code into svg files. The svg files will be found in ~\src\svg  
directory.

### CODE REVIEW

File: src/functions.js [hunk 1]
```diff
@@ -1,51 +1,74 @@
...
+  });
+};
+
+export const writeSvgCodeIntoSvgFiles = directory => {
+  const fileNames = getAllFileNames(directory);
+  fileNames.forEach(fileName => {
+    var filePath = directory + fileName;
+    var oldData = getFileData(filePath).toString();
+    if (oldData.search(CODE_ELEMENT) < 0) {
+      var PositionToInsert = oldData.indexOf("</svg>");
```
Comment (CastelloDev) [inline src/functions.js]: does this not need a +1

File: src/functions.js [hunk 2]
```diff
@@ -1,51 +1,74 @@
...
+      var newData = [
+        oldData.slice(0, PositionToInsert),
+        CODE_ELEMENT,
+        oldData.slice(PositionToInsert, oldData.length)
+      ].join("");
+
+      fs.writeFile(filePath, "\n" + newData + "\n", err => {
```
Comment (CastelloDev) [inline src/functions.js]: do you have to add newline arguements for this to work?

**Review by CastelloDev — CHANGES_REQUESTED** [review]

File: src/functions.js [hunk 3]
```diff
@@ -1,51 +1,74 @@
...
+        if (err) console.log(err);
+        console.log("Successfully Written to File.");
```
Comment (CastelloDev) [inline src/functions.js]: the way you wrote this will display Successfully written to file even if there has been an error

**Review by CastelloDev — CHANGES_REQUESTED** [review]

File: src/functions.js (hunk 1 above)
Comment (CastelloDev) [reply src/functions.js]: if it doesnt, please send me a text snippet of an entire svg file that has been processed with your code

Comment (SniraJavas) [reply src/functions.js]: Alright, I have discovered that I should add my code before </svg> no longer immediately after <svg>.

Comment (SniraJavas) [reply src/functions.js]: imagine a file that start with a lot of comments or something like the following statement :```
 <?xml version="1.0" encoding="utf-8"?>
 
``` ,

In cant use the first ">" as there might be many ">"  before I reach the that is enclosing svg. 

There other approach will always work.

File: src/functions.js (hunk 2 above)
Comment (SniraJavas) [reply src/functions.js]: New line was for styling so it can look nice, but with or without it can work fine.

File: src/functions.js (hunk 3 above)
Comment (SniraJavas) [reply src/functions.js]: Okay, I forgot to return that error !

File: src/functions.js (hunk 1 above)
Comment (CastelloDev) [reply src/functions.js]: it should be at the very top of the svg
svgomg should remove all def tags and style tags as well as all comments, so inserting it athe top should be easy

File: src/functions.js (hunk 2 above)
Comment (CastelloDev) [reply src/functions.js]: cool, remember we're trying to optimize this, so any unncessary characters will only make the svg bigger than it could be

Comment (SniraJavas) [reply src/functions.js]: alright, I get you !

File: src/functions.js (hunk 1 above)
Comment (SniraJavas) [reply src/functions.js]: Alright, DOne

File: src/functions.js [hunk 4]
```diff
@@ -1,51 +1,78 @@
...
+    if (oldData.search(CODE_ELEMENT) < 0) {
+      var PositionToInsert = oldData.indexOf(">")+1;
+      var newData = [
+        oldData.slice(0, PositionToInsert),
+        CODE_ELEMENT,
+        oldData.slice(PositionToInsert, oldData.length)
+      ].join("");
+
+      fs.writeFile(filePath,newData, err => {
+        if (!err) {
```
Comment (CastelloDev) [inline src/functions.js]: err is probably a string, switch your logic around, if(err) checks if err exists, then is handled accordingly, please stick to this way of handling errors

File: src/functions.js [hunk 5]
```diff
@@ -1,51 +1,78 @@
...
+          console.log("Successfully Written to File.");
```
Comment (CastelloDev) [inline src/functions.js]: please log the name of the file that has been saved, more meaningful that way

**Review by CastelloDev — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -19,7 +19,8 @@
   "homepage": "https://github.com/CastelloDev/svg-optimizer.git#readme",
   "dependencies": {
     "esm": "3.2.25",
-    "simple-svg-tools": "1.1.12",
-    "svgo": "1.3.0"
+    "fs": "0.0.1-security",
+    "i": "^0.3.6",
```
Comment (CastelloDev) [inline package.json]: whats does this dependency do?

File: package.json [hunk 7]
```diff
@@ -19,7 +19,8 @@
...
+    "simple-svg-tools": "^1.1.12"
```
Comment (CastelloDev) [inline package.json]: please remove all ^

**Review by CastelloDev — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (SniraJavas) [reply package.json]: DOne

File: package.json (hunk 6 above)
Comment (SniraJavas) [reply package.json]: removed i dependancy

File: package.json [hunk 8]
```diff
@@ -19,7 +19,7 @@
   "homepage": "https://github.com/CastelloDev/svg-optimizer.git#readme",
   "dependencies": {
     "esm": "3.2.25",
-    "simple-svg-tools": "1.1.12",
-    "svgo": "1.3.0"
```
Comment (CastelloDev) [inline package.json:23]: add this dependency back please

**Review by CastelloDev — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (SniraJavas) [reply package.json:23]: did

**Review by hvandermerwe — APPROVED** [review]

**Review by CastelloDev — APPROVED** [review]

**Pull request merged**